pkg/ipcache/metrics.go

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ipcache
+
+var (
+	metricTypeUpsert  = "upsert"
+	metricTypeDelete  = "delete"
+	metricTypeRecover = "recover"
+
+	metricErrorIdempotent = "idempotent_operation"
+	metricErrorInvalid    = "invalid_prefix"
+	metricErrorNoExist    = "no_such_prefix"
+	metricErrorOverwrite  = "cannot_overwrite_by_source"
+	metricErrorUnexpected = "upsert_unexpectedly_deleted_entry"
```

Comment:
do we have any success one as the metric name is `ipcache_events_total`, but not `ipcache_error_events_total` ?

Reply:
Ah, so far I was just doing `error=""` but I can change that to `error="success"` if we think that's easier to understand. What do you think?

Reply:
error="none" seems more more friendly I feel :thinking:

Reply:
having error as a label seems a bit odd to me, especially when it can be empty in some cases. Would something like type="<event_type>" make more sense?

Reply:
I see we already follow this pattern for other metrics however. So that might be breaking with convention.

Reply:
Something like this seems more intuitive: https://github.com/cilium/cilium/blob/b92ce4afc1bd29316373e831c25efea020786f2c/pkg/metrics/metrics.go#L806

Reply:
@tommyp1ckles are you suggesting that error conditions should be a whole different metric? This could be a way around it. Rather than adding `cilium_ipcache_events_total{error=foo}`, we could have separately `cilium_ipcache_events_total` and `cilium_ipcache_errors_total{error=foo}`?

(Though I think in some other parts of the codebase, we keep them together to avoid doing something like `if error { myErrorMetric.Inc() } else {myTotalMetric.Inc() }`.

Reply:
> having error as a label seems a bit odd to me, especially when it can be empty in some cases. Would something like type="<event_type>" make more sense?

I guess this is another angle on it.. but even with `type="<event_type>"`, we would still have `error=foo` right? As long as the errors are a predefined set. We would still need `error=none` for `type=success`.

Reply:
Some wisdom from @chancez on [my other PR](https://github.com/cilium/cilium/pull/20753#pullrequestreview-1060516925):

> If you can't sum the metric over all dimensions and get something useful, then it indicates that things should be broken apart, which seems to be the case here.

Reply:
I'll split the metrics up into `ipcache_errors_total` vs. `ipcache_events_total` in my upcoming rebase.